packages/valory/skills/twitter_scoring_abci/rounds.py

```diff
@@ -615,9 +615,9 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
             ).performed_twitter_tasks
             performed_twitter_tasks["db_update"] = Event.DONE.value
 
-            # Clear processed tweets that are no longer needed
+            # Clear processed tweets that are no longer needed. Keep only those with no points yet.
             tweets = cast(SynchronizedData, self.synchronized_data).tweets
-            tweets = {k: v for k, v in tweets.items() if "points" in v}
+            tweets = {k: v for k, v in tweets.items() if "points" not in v}
```

Comment:
This is the fix. While introducing the mech into IEKit, on commit `205c7a65b95e01cc1ac354424cc0c0f4c5c30704` I forgot the "not" while refactoring.

Reply:
Here's the commit where the bug was introduced: https://github.com/valory-xyz/IEKit/pull/99/commits/205c7a65b95e01cc1ac354424cc0c0f4c5c30704#r1452300512